addons/mail/models/mail_message.py

```diff
@@ -659,9 +659,22 @@ def unlink(self):
         self.mapped('attachment_ids').filtered(
             lambda attach: attach.res_model == self._name and (attach.res_id in self.ids or attach.res_id == 0)
         ).unlink()
+        messages_by_partner_with_user = defaultdict(lambda: self.env['mail.message'])
```

Comment:
Would rename "messages_by_partner", the "with_user" part is a bit of a detail.

Reply:
Changed, it was to help understand (that it was filtered by partner having a user) but indeed long name can also be counter-productive.